### PR TITLE
[SWE-14] Login Page

### PR DESCRIPTION
**Pull request by sammyoto**
Created a basic login page. Also added a public folder with a website favicon.

### CODE REVIEW

File: pages/login.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    let username = formDataObj["username"]
+    let password = formDataObj["password"]
+
+    // Check if user input username and password
+    if (username.length == 0 || password.length == 0) {
+      let error = document.getElementById('error-message');
+      error.innerText = "Please enter both a Username and Password."
+      return
+    }
```
Comment (datderek) [inline pages/login.js:22]: May be better to re-render the login component by using with [React states](https://legacy.reactjs.org/docs/hooks-state.html)? Check out the `useState `to update the `errors` variable in the `create-account.js`. What are your thoughts @jackfales?

File: public/favicon-32x32.png
Comment (datderek) [inline public/favicon-32x32.png:1]: Thoughts on changing the favicon to just the 'U' (smiley face with tongue sticking out) part of the YUM?

**Review by datderek — CHANGES_REQUESTED** [review]
Had a question on how should handle re-rendering, otherwise LGTM.

File: public/favicon-32x32.png
Comment (datderek) [reply public/favicon-32x32.png:1]: I can give you that image if we decide to go down that route.

File: pages/login.js (hunk 1 above)
Comment (sammyoto) [reply pages/login.js]: just fixed this. should be good now.

**Review by datderek — APPROVED** [review]
LGTM

**Comment by jackfales** [comment]
1. This is [SWE-15](https://linear.app/swe-dem-boys/issue/SWE-15/login-page-frontend) not SWE-14. Change the PR name and branch name accordingly.
2. Your task name is **Login Page Frontend** not **Login Page**. It's important these are distinct because the Login Page is an entire component while this PR only covers the frontend. Change accordingly.

File: package-lock.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "yum",
+  "name": "recipe-tracker",
```
Comment (jackfales) [inline package-lock.json:2]: Why did you change this? It should be `yum`

File: pages/login.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    let password = formDataObj["password"]
+
+    // Check if user input username and password
+    if (username.length == 0 || password.length == 0) {
+      setError("Please enter both a Username and Password.")
+      return
+    }
+    else {
+      setError("")
+    }
```
Comment (jackfales) [inline pages/login.js:26]: nit: `else` blocks should begin on the same line as the end curly brace of an `if` block
```suggestion
    } else {
      setError("")
    }
```

File: pages/login.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  function onSubmit(event) {
+    event.preventDefault()
+
+    // Get form data and create JSON object
+    const formData = new FormData(event.target)
+    const formDataObj = {}
+    formData.forEach((value, key) => (formDataObj[key] = value))
+
+    let username = formDataObj["username"]
+    let password = formDataObj["password"]
```
Comment (jackfales) [inline pages/login.js:17]: Why are you putting `formData` into a separate object? Can't we just iterate over `formData` instead?

File: pages/login.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    if (username.length == 0 || password.length == 0) {
+      setError("Please enter both a Username and Password.")
+      return
+    }
+    else {
+      setError("")
+    }
+
+    console.log(username)
+    console.log(password)
```
Comment (jackfales) [inline pages/login.js:29]: Helps with logging readability

```suggestion
    console.log("Username: %s", username)
    console.log("Password: %s", password)
```

File: pages/login.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
+import Head from 'next/head';
+import { useState } from 'react';
+import styles from '../styles/Home.module.css';
+
+export default function Login() {
+  const [error, setError] = useState('');
```
Comment (jackfales) [inline pages/login.js:6]: nit:
```suggestion
  const [errorMessage, setErrorMessage] = useState('');
```

**Review by jackfales — CHANGES_REQUESTED** [review]
Overall, good job. Few changes and questions to address as well as changing the branch and PR name for consistency with our task log.
